charger/nrggen2_json.go

```diff
@@ -0,0 +1,314 @@
+	"github.com/evcc-io/evcc/util/transport"
+)
+
+// https://www.nrgkick.com/wp-content/uploads/2024/07/local_api_docu_simulate.html
+
+// NRGKickGen2 charger implementation
+type NRGKickGen2 struct {
+	*request.Helper
+	uri      string
+	password string
+	enabled  bool
+	controlG provider.Cacheable[gen2.Control]
+	valuesG  provider.Cacheable[gen2.Values]
+	infoG    provider.Cacheable[gen2.Info]
+}
+
+func init() {
+	registry.Add("nrgkick-gen2", NewNRGKickGen2FromConfig)
+}
+
+// NewNRGKickGen2FromConfig creates a NRGKickGen2 charger from generic config
+func NewNRGKickGen2FromConfig(other map[string]interface{}) (api.Charger, error) {
+	cc := struct {
+		URI, User, Password string
+		Cache               time.Duration
+	}{
+		Cache: time.Second,
+	}
+
+	if err := util.DecodeOther(other, &cc); err != nil {
+		return nil, err
+	}
+
+	return NewNRGKickGen2(cc.URI, cc.User, cc.Password, cc.Cache)
+}
+
+// NewNRGKickGen2 creates NRGKickGen2 charger
+func NewNRGKickGen2(uri, user, password string, cache time.Duration) (*NRGKickGen2, error) {
+	basicAuth := transport.BasicAuthHeader(user, password)
+	log := util.NewLogger("nrggen2").Redact(user, password, basicAuth)
+
+	nrg := &NRGKickGen2{
+		Helper:   request.NewHelper(log),
+		uri:      util.DefaultScheme(uri, "http"),
+		password: password,
+	}
+
+	if user != "" && password != "" {
+		nrg.Client.Transport = transport.BasicAuth(user, password, nrg.Client.Transport)
+	}
+
+	nrg.controlG = provider.ResettableCached(func() (gen2.Control, error) {
+		var res gen2.Control
+
+		err := nrg.GetJSON(nrg.apiURL(gen2.ControlPath), &res)
+		if err != nil && res.Response != "" {
+			err = errors.New(res.Response)
+		}
+
+		return res, err
+	}, cache)
+
+	nrg.valuesG = provider.ResettableCached(func() (gen2.Values, error) {
+		var res gen2.Values
+
+		err := nrg.GetJSON(nrg.apiURL(gen2.ValuesPath), &res)
+
+		return res, err
+	}, cache)
+
+	nrg.infoG = provider.ResettableCached(func() (gen2.Info, error) {
+		var res gen2.Info
+
+		err := nrg.GetJSON(nrg.apiURL(gen2.InfoPath), &res)
+
+		return res, err
+	}, cache)
+
+	return nrg, nil
+}
+
+func (nrg *NRGKickGen2) apiURL(api string) string {
+	return fmt.Sprintf("%s/%s", nrg.uri, api)
+}
+
+func (nrg *NRGKickGen2) updateControl(control gen2.Control, withPhaseSwitch bool) error {
+	// TODO: i am not sure if we can set current_set with decimals over the query parameter, so just in case i limit it to the integer value
+	uriWithQueryParams := fmt.Sprintf(
+		"%s?current_set=%.0f&charge_pause=%d&energy_limit=%d",
+		nrg.apiURL(gen2.ControlPath),
+		control.CurrentSet,
+		control.ChargePause,
+		control.EnergyLimit)
+
+	if withPhaseSwitch {
+		uriWithQueryParams = fmt.Sprintf("%s&phase_count=%d", uriWithQueryParams, control.PhaseCount)
+	}
+
+	req, err := request.New(http.MethodGet, uriWithQueryParams, nil, request.AcceptJSON)
+	if err != nil {
+		return err
+	}
+
+	var res gen2.Control
+
+	if err := nrg.DoJSON(req, &res); err != nil {
```

Comment:
Why not use `GetJSON` similar to https://github.com/evcc-io/evcc/pull/14938/files#diff-e66d191361cf8ba2a343492ad6d253ad448eddd02ab4c4630df3cf3478c4084dR69?